src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -1219,6 +1252,65 @@ internal async void OnTimer(object sender, System.Timers.ElapsedEventArgs e)
             }
         }
 
+        internal void OnInvocationTimer(object sender, System.Timers.ElapsedEventArgs e)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            try
+            {
+                CheckAndRecycleFaultyLanguageWorker();
+            }
+            catch
+            {
+                // Don't allow background execptions to escape
+                // E.g. when a rpc channel is shutting down we can process exceptions
+            }
+            try
+            {
+                _invocationTimer.Start();
+            }
+            catch (ObjectDisposedException)
+            {
+                // Specifically ignore this race - we're exiting and that's okay
+            }
+        }
+
+        internal void CheckAndRecycleFaultyLanguageWorker()
+        {
+            _workerChannelLogger.LogInformation("Checking and recycling faulty language workers");
+
+            Dictionary<string, int> failureRatePerWorkerId = new Dictionary<string, int>();
+            var failureThreshold = 30; // ToDo: Moving this constant once finalized
+
+            foreach (var item in _invocationMetricsPerWorkerId)
+            {
+                int failureRateOfWorker = item.Value.TotalInvocations > 0 ? (item.Value.TotalInvocations - item.Value.SuccessfulInvocations) * 100 / item.Value.TotalInvocations : 0;
+                failureRatePerWorkerId[item.Key] = failureRateOfWorker;
+
+                // To be removed
+                _workerChannelLogger.LogInformation($"WorkerId: {item.Key}, Failure rate: {failureRateOfWorker}");
+            }
+
+            var sortedFailureRatePerWorkerId = from item in failureRatePerWorkerId orderby item.Value descending select item;
+            var workerIdToRecycle = sortedFailureRatePerWorkerId.First().Key;
+            var failureRate = failureRatePerWorkerId[workerIdToRecycle];
+
+            if (failureRate > failureThreshold)
+            {
+                // ToDo: in next iteration of this feature => recycle the workerId in variable: workerIdToRecycle
+                // ToDo: Other criteria for recycle => if invocation latency is too high
+
+                _metricsLogger.LogEvent(string.Format(MetricEventNames.WorkerRecycled, Id), data: $"WorkerId to be recycled: {workerIdToRecycle}");
+                _workerChannelLogger.LogDebug($"WorkerId to be Recycled: {workerIdToRecycle}, Failure rate: {failureRate}");
```

Comment:
Please us structured logging.
```suggestion
                _workerChannelLogger.LogDebug("WorkerId to be Recycled: {workerId}, Failure rate: {failureRate}", workerIdToRecycle, failureRate);
```

Reply:
Done